src/storage-client/src/source/persist_source.rs

```diff
@@ -159,111 +165,132 @@ where
     let mut row_builder = Row::default();
 
     // Extract the MFP if it exists; leave behind an identity MFP in that case.
-    let mut map_filter_project = map_filter_project.as_mut().map(|mfp| mfp.take());
+    let map_filter_project = map_filter_project.as_mut().map(|mfp| mfp.take());
 
-    builder.build(move |mut initial_capabilities| async move {
-        initial_capabilities.clear();
+    builder.build(move |_caps| {
+        // Acquire an activator to reschedule the operator when it has unfinished work.
+        let activations = scope.activations();
+        let activator = Activator::new(&operator_info.address[..], activations);
+        // Maintain a list of work to do
+        let mut todo = std::collections::VecDeque::new();
```

Comment:
super nit: `TODO` comments are common enough that this variable name gave me pause while reading the new code. maybe call it `pending_work`?